app/uk/gov/hmrc/cataloguefrontend/shuttering/ShutterConnector.scala

```diff
@@ -117,3 +124,16 @@ class ShutterConnector @Inject()(
       .GET[Seq[FrontendRouteWarning]](s"${urlFrontendRouteWarnings(env)}/${encodePathParam(serviceName)}")
   }
 }
+
+object ShutterConnector {
+  case class ShutterEventsFilter(environment: Environment, serviceName: Option[String])
+
+  object ShutterEventsFilter {
+    private [shuttering] def asQuery(filter: ShutterEventsFilter): String =
+      asQueryParam(key = "data.environment", value = filter.environment.asString) +
```

Comment:
maybe move `&` out of queryParam and into join e.g. `(Seq(queryParam("data.environment", environment) + serviceName.map(queryParam("data.serviceName", _)).join("&")`
This will move the initial `&` into the url (where we have context of if a previous query param has been used)

Reply:
Agree that seems cleaner.